src/borg/archive.py

```diff
@@ -1052,6 +1052,19 @@ def verify_data(self):
                 self.error_found = True
                 errors += 1
                 logger.error('chunk %s, integrity error: %s', bin_to_hex(chunk_id), integrity_error)
+                if self.repair:
+                    # if we kill the defect chunk here, subsequent actions within this "borg check"
+                    # run will find missing chunks and replace them with all-zero replacement
+                    # chunks and flag the files as "repaired".
+                    # if another backup is done later and the missing chunks get backupped again,
+                    # a "borg check" afterwards can heal all files where this chunk was missing.
+                    logger.warning('chunk %s is being deleted, so affected files can get repaired now '
+                                   'and maybe healed later.', bin_to_hex(chunk_id))
+                    del self.chunks[chunk_id]
+                    self.repository.delete(chunk_id)
+                else:
+                    logger.warning('chunk %s would be deleted, so affected files can get repaired now '
+                                   'and maybe healed later.', bin_to_hex(chunk_id))
```

Comment:
This message is kinda confusing. `--repair would delete the corrupted chunk, so affected files could be repaired (and may be healed later)` or so. Both of these warnings should only be printed once, since there already is a per-chunk message and they just clarify what will happen / would happen.